### DISCUSSION TITLE
How does persistence work?

### DISCUSSION THREAD

**Discussion by disto** [CodeDredd/pinia-orm#1151] [Q&A] [answered]
Hi there, 
Thanks for making this beautiful feature!
I try to persist the data in the orm. I does not seem to work, perhaps I misunderstand what persistence is.
I expect the data in the ORM to be saved to local storage (session storage), however despite me setting this setting
```
  static piniaOptions = {
    persist: true
  }
```
on my users model, refreshing the browser clears the store. 
Please advice how to store my users data.

Kind regards

**Comment by CodeDredd** [comment, marked as answer]
Have you installed https://github.com/prazdevs/pinia-plugin-persistedstate ?

**Comment by treeindark** [comment]
Yes i do that! but , when i reload my store is not rehydrate from store:

dataRepo.all() -> []

**Comment by MangMax** [comment]
I installed it, but I don't know how it works together, it doesn't seem to work

**Reply by MangMax** [reply]
I tried this and it works, but it uses cookies by default!
```ts
  static piniaOptions = {
    persist:true
  }
```
When I try to configure it this way it doesn't work
```ts
static piniaOptions ={
	persist: {
	      storage:'localStorage',
	      serializer: {
	        deserialize: parse,
	        serialize: stringify,
	      },
	    }
}
```
